importlab/resolve.py

```diff
@@ -26,11 +27,58 @@ def __init__(self, module_name):
         self.module_name = module_name
 
 
-class ModuleAndDependencies(object):
-    def __init__(self, filename):
-        self.filename = filename
-        self.name, self.ext = os.path.splitext(os.path.basename(self.filename))
-        self.directory = os.path.dirname(self.filename)
+class ResolvedFile(object):
+    def __init__(self, path, module_name):
+        self.path = path
+        self.module_name = module_name
+
+    def is_extension(self):
+        return self.path.endswith('.so')
+
+    @property
+    def short_path(self):
```

Comment:
What exactly is the short path? For, e.g., a path of `/usr/lib/python3/dist-packages/urllib3/poolmanager.py` and a module name of `..poolmanager`, I'm seeing a short path of `dist-packages/urllib3/poolmanager.py`, and I'm not sure what to make of that conceptually.